src/main/kotlin/no/nav/familie/ks/kontrakter/oppdrag/Utbetalingsoppdrag.kt

```diff
@@ -0,0 +1,38 @@
+package no.nav.familie.ks.kontrakter.oppdrag
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+data class Utbetalingsoppdrag (
+    val kodeEndring: KodeEndring,
+    val fagSystem: String,
+    val saksnummer: String,
+    val aktoer: String,
+    val saksbehandlerId: String,
+    val avstemmingTidspunkt: LocalDateTime = LocalDateTime.now(),
+    val utbetalingsperiode: List<Utbetalingsperiode>
+) {
+    enum class KodeEndring { NY, ENDR, UENDR }
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class Utbetalingsperiode (
+    val kodeEndringPeriode: KodeEndringPeriode,
+    val statusPeriode: StatusPeriode?,
+    val statusDatoFom: LocalDate,
```

Comment:
Dette kan kanskje også omformuleres, feks som en boolean `erOpphør` og `opphørFom`? :) Og skal dato-feltet også være optional? Ser ut som det kun settes når statusPeriode-feltet er satt. Mulig vi kan lage en egen type for dette som er optional, typ noe sånt: 
```
val opphør: Opphør?
...
data class Opphør (val erOpphør: boolean, val opphørDatoFom: LocalDate)
```
Da får vi en garanti for at datoen er satt i alle tilfeller hvor man setter opphør. Mulig at man også bare trenger dato-feltet, og hvis det er satt, så vet vi at vi skal ha status opphør i Oppdrag110-objektet.

Reply:
Særdeles godt tenkt!!